compiler-core/src/type_/error.rs

```diff
@@ -789,6 +791,12 @@ pub enum UnknownValueConstructorError {
     },
 }
 
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum UnknownValueConstructorErrorSituation {
+    /// A value using ```let``` is assigned to a variable having a capital case.
+    NotFoundPattern,
```

Comment:
The name of this variant doesn't match its meaning or the documentation above. Let's give a descriptive and accurate name.